plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -551,9 +555,9 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         DeltaLakeColumnHandle column = (DeltaLakeColumnHandle) columnHandle;
         return getColumnMetadata(
                 column,
-                getColumnComments(table.getMetadataEntry()).get(column.getName()),
-                getColumnsNullability(table.getMetadataEntry()).getOrDefault(column.getName(), true),
-                getGeneratedColumnExpressions(table.getMetadataEntry()).get(column.getName()));
+                getColumnComments(table.getMetadataEntry()).get(column.getBaseColumnName()),
+                getColumnsNullability(table.getMetadataEntry()).getOrDefault(column.getBaseColumnName(), true),
+                getGeneratedColumnExpressions(table.getMetadataEntry()).get(column.getBaseColumnName()));
```

Comment:
this is called in `io.trino.sql.planner.InputExtractor.Visitor#processScan`
so `column` may represent a dereference 
returnning comment, nullability or generating expression of the base column isn't correct

Reply:
I'm guessing that we should probably add a check here for the case when the column handle has `projectionInfo` present and use `null, true, null` in getting the column metadata.